crates/telio-proxy/src/proxy.rs

```diff
@@ -230,12 +230,10 @@ impl Runtime for StateIngress {
             wait_for_tx(&self.output, select_all(futures)).await
         {
             if let Ok(n) = socket.try_recv(self.read_buf.as_mut_slice()) {
-                let msg = DataMsg::new(if let Some(buf) = self.read_buf.get(..n) {
-                    buf
-                } else {
-                    return Self::error(());
-                });
-                let _ = permit.send((pk, msg));
+                if let Some(buf) = self.read_buf.get(..n) {
+                    let msg = DataMsg::new(buf);
+                    let _ = permit.send((pk, msg));
+                }
```

Comment:
So get it self I don't think as an issue, it should not really be possible to overstep n, as it will always be smaller that read_buf.

But probably a better question is what happens if socket for some reason dies.
It may cause a busy loop, and we should probably decide what we should do in such a case.

Reply:
Created LLT-4406 to address this